embedded/document/document_id.go

```diff
@@ -0,0 +1,90 @@
+package document
+
+import (
+	"crypto/rand"
+	"encoding/binary"
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"io"
+	"time"
+)
+
+/*
+	DocumentID is a 16-byte identifier that is automatically generated
+	by the server upon document insertion. The 16-byte DocumentID is composed of:
+
+		1) A 4-byte timestamp value, representing the time the document was created,
+		measured in seconds since the Unix epoch.
+		2) A 4-byte random value, generated using a secure random number generator.
+		3) A 8-byte value, initialized to the previous transaction id.
```

Comment:
using 8-bytes for timestamp might be better

it may be better to include txid after timestamp

Reply:
it might also be good to suffix with a monotically increasing number, such number may be restarted when immudb starts or when the txid fragment changes. In such case, not sure of the need of the random piece...

Reply:
fixed, added monotically increasing number. Time stamp I feel can be 4 bytes